src/string/stpncpy.c

```diff
@@ -7,9 +7,15 @@
 #define HIGHS (ONES * (UCHAR_MAX/2+1))
 #define HASZERO(x) ((x)-ONES & ~(x) & HIGHS)
 
-char *__stpncpy(char *restrict d, const char *restrict s, size_t n)
-{
+char *__stpncpy(char *restrict d : itype(restrict _Nt_array_ptr<char>) count(n),
+                const char *restrict s : itype(restrict _Nt_array_ptr<const char>) count(n),
+                size_t n)
+  :itype(_Nt_array_ptr<char>) count(n)
+_Checked{
+// This part is GCC Specific code and uses unchecked pointer,
```

Comment:
Actually, this does not seem to be GCC-specific code. This code optimizes memcpy if the src and dest alias. Without, this code it falls-back to the naive byte-by-byte copy. What error do you get if clang compiles this code?

See the following patch which introduces the GNUC check:
```
commit 4d0a82170a25464c39522d7190b9fe302045ddb2
Author: Rich Felker <dalias@aerifal.cx>
Date:   Wed Sep 26 14:39:10 2018 -0400

    fix aliasing-based undefined behavior in string functions

    use the GNU C may_alias attribute if available, and fallback to naive
    byte-by-byte loops if __GNUC__ is not defined.

    this patch has been written to minimize changes so that history
    remains reviewable; it does not attempt to bring the affected code
    into a more consistent or elegant form.
```